I2CEEBlockDevice.cpp

```diff
@@ -21,11 +21,26 @@
 I2CEEBlockDevice::I2CEEBlockDevice(
         PinName sda, PinName scl, uint8_t addr,
         bd_size_t size, bd_size_t block, int freq)
-    : _i2c(sda, scl), _i2c_addr(addr), _size(size), _block(block)
+    : _i2c_p(new I2C(sda, scl)), _i2c(*_i2c_p), _i2c_addr(addr),
```

Comment:
Could you instead use [placement new](https://en.wikipedia.org/wiki/Placement_syntax)? That way the class still uses the memory needed for the I2C object and we don't have to rely on dynamic memory.

``` c++
uint32_t _i2c_buffer[sizeof(I2C) / sizeof(uint32_t);
I2C *_i2c;
_i2c = new (_i2c_buffer) I2C(sda, scl);
_i2c->~I2C(); // note needs manual delete
```